File: web-admin-front/src/mock/movie.ts
import type { MovieItem } from '../types/movie';

export const mockMovies: MovieItem[] = [
  {
    id: 1, name: '流浪地球3', types: ['scifi', 'action'],
    typeLabel: '科幻、动作', poster_url: '', rating: 9.2, duration: 173,
    release_date: '2026-08-15', director: '郭帆', actors: '吴京、刘德华',
    description: '太阳即将毁灭，人类在地球表面建造出巨大的推进器，寻找新的家园。',
    status: 'showing', hasSchedule: true,
  },
  {
    id: 2, name: '哪吒之魔童闹海', types: ['animation', 'action'],
    typeLabel: '动画、动作', poster_url: '', rating: 8.8, duration: 144,
    release_date: '2026-07-20', director: '饺子', actors: '',
    description: '哪吒与敖丙天劫之后，太乙真人用七色宝莲重塑肉身的故事。',
    status: 'showing', hasSchedule: true,
  },
  {
    id: 3, name: '唐人街探案4', types: ['comedy', 'suspense'],
    typeLabel: '喜剧、悬疑', poster_url: '', rating: 7.5, duration: 132,
    release_date: '2026-08-01', director: '陈思诚', actors: '王宝强、刘昊然',
    description: '唐仁和秦风受邀前往伦敦，揭开一起跨国奇案。',
    status: 'showing', hasSchedule: false,
  },
  {
    id: 4, name: '封神第二部', types: ['action', 'scifi'],
    typeLabel: '动作、科幻', poster_url: '', rating: 8.3, duration: 150,
    release_date: '2026-07-15', director: '乌尔善', actors: '费翔、黄渤',
    description: '姬发回到西岐，殷商大军压境，一场大战即将爆发。',
    status: 'showing', hasSchedule: true,
  },
  {
    id: 5, name: '熊出没·重启未来', types: ['animation', 'comedy'],
    typeLabel: '动画、喜剧', poster_url: '', rating: 7.8, duration: 108,
    release_date: '2026-08-10', director: '丁亮', actors: '',
    description: '光头强和熊大熊二意外穿越到未来世界。',
    status: 'offline',
  },
  {
    id: 6, name: '热辣滚烫2', types: ['comedy', 'romance'],
    typeLabel: '喜剧、爱情', poster_url: '', rating: 7.2, duration: 129,
    release_date: '2026-07-28', director: '贾玲', actors: '贾玲、雷佳音',
    description: '乐莹拳击归来，面对新的生活挑战。',
    status: 'showing',
  },
  {
    id: 7, name: '志愿军：存亡之战', types: ['action', 'other'],
    typeLabel: '动作、其他', poster_url: '', rating: 8.1, duration: 155,
    release_date: '2026-07-10', director: '陈凯歌', actors: '唐国强、刘劲',
    description: '讲述抗美援朝战争中铁原阻击战的故事。',
    status: 'showing', hasSchedule: true,
  },
  {
    id: 8, name: '深海2', types: ['animation', 'suspense'],
    typeLabel: '动画、悬疑', poster_url: '', rating: 8.6, duration: 122,
    release_date: '2026-08-20', director: '田晓鹏', actors: '',
    description: '参宿再次坠入深海世界，这一次她要寻找回家的路。',
    status: 'offline',
  },
];

File: ticket-service/src/main/java/org/dherhf/movie/entity/Movie.java
package org.dherhf.movie.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 影片实体类，对应数据库 movie 表。
 * <p>
 * 存储影片基本信息，包括名称、类型、海报、评分、时长、上映日期、导演、演员等。
 * 使用 MyBatis-Plus 注解进行 ORM 映射，{@code types} 字段通过 JacksonTypeHandler 存为 JSON。
 * 逻辑删除字段 {@code deleted} 实现软删除。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "movie", autoResultMap = true)
public class Movie {

    /** 主键 ID，雪花算法生成 */
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /** 影片名称 */
    private String name;

    /** 影片类型列表，JSON 存储（如动作、科幻、喜剧） */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> types;

    /** 海报图片在 OSS 中的 objectKey */
    private String posterUrl;

    /** 影片评分 */
    private BigDecimal rating;

    /** 影片时长（分钟） */
    private Integer duration;

    /** 上映日期 */
    private LocalDate releaseDate;

    /** 导演 */
    private String director;

    /** 演员 */
    private String actors;

    /** 影片简介 */
    private String description;

    /** 影片状态：0-下架，1-上架 */
    private Integer status;

    /** 逻辑删除标识：0-未删除，1-已删除 */
    @TableLogic
    private Integer deleted;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
